File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    nums = [ i * (i + 1) // 2 for i in range(n//2 + 2)]

    answer = 0
    for right in range(1, n//2 + 2):
        left = right - 1
        while left > 0 and nums[right] - nums[left] < n:
            left -= 1
            
        if nums[right] - nums[left] == n:
            answer += 1

    
    return answer + 1 if n > 2 else answer
            

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int[] nums = new int[n/2 + 2];
        for (int i = 1; i < n/2 + 2; i++) {
            nums[i] = i * (i + 1) / 2;
        }
        int answer = 0;
        
        for(int right = 1; right < n/2 + 2; right++) {
            int left = right - 1;
            while(left > 0 && nums[right] - nums[left] < n) {
                left--;
            }
            
            if (nums[right] - nums[left] == n) {
                answer++;
            }
        }
        
        return n > 2 ? answer + 1 : answer;
    }
}